docs/content/design/segments.md

```diff
@@ -4,7 +4,120 @@ layout: doc_page
 Segments
 ========
 
-Druid segments contain data for a time interval, stored as separate columns. Dimensions (string columns) have inverted indexes associated with them for each dimension value. Metric columns are LZ4 compressed.
+
+Druid stores its index in *segment files*. In a basic setup, one
+segment file is created for each time interval, where the time inteval
+is configurable in the indexing spec.  For druid to operate well under
+heavy query load, it is important for the segment file size to be
+within the recommended range of 300mb-700mb. If your segment files are
+larger than this range, then consider either changing the the
+granularity of the time interval or partitioning your data and
+tweaking the `targetPartitionSize` in your `partitioningSpec` (a good
+starting point for this parameter is 5 million rows).  See the section
+below on sharding or the documentation on indexing for more information.
```

Comment:
it's all flattened on the github side.  If you look in the "_bin" folder, you can see the scripts that we use to generate PRs against the druid-io.github.io branch.